### ISSUE TITLE
What `Content-Type` should be used?

### ISSUE THREAD

**Issue by Kludex** [build-server-protocol/build-server-protocol#265]
The [Content-Type: "application/vscode-jsonrpc"](https://microsoft.github.io/language-server-protocol/specification#headerPart) is defined for LSP, even if [not registered](https://www.iana.org/assignments/media-types/media-types.xhtml).

What should I do here?

**Comment by jastice** [comment]
Unspecified, and at the moment clients ignore this. Let's go with `application/json-rpc` as per the [JSON-RPC spec](https://www.jsonrpc.org/historical/json-rpc-over-http.html#http-header):

> Content-Type SHOULD be 'application/json-rpc' but MAY be 'application/json' or 'application/jsonrequest'

**Comment by Kludex** [comment]
That's jsonrpc 1.0. On 2.0 is `application/json`.

I'm on my phone, so I'll not give reference.

**Comment by jastice** [comment]
Good point. On the website it looks like it's [not specified](https://www.jsonrpc.org/specification) but `application/json` seems to be what people are using so let's go with that.

**Comment by Kludex** [comment]
About what I said: https://www.simple-is-better.org/json-rpc/transport_http.html

I guess this last link is more up-to-date (look at the dates), but your link looks more _official_ :thinking:
